externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -514,4 +515,41 @@ def Tcp_Atan2Op : Tcp_BinaryElementwiseOp<"atan2", [SameOperandsAndResultElement
   let assemblyFormat = "$in1 `,` $in2 attr-dict `:` type($in1) `,` type($in2) `->` type($out)";
 }
 
+def Tcp_CastOp:  Tcp_Op<"cast", [Pure, Elementwise, SameOperandsAndResultShape]> {
+
+  let summary = "TCP Cast operation";
+
+  let description = [{
+    Performs elementwise type conversion.
+
+    Supported torch/mlir element type:
+      INT: ui8,  si8,  i8, 
+           ui16, si16, i16,
+           ui32, si32, i32,
+           ui64, si64, i64, 
+           i1 (Boolean)
+      FP: f16, bf16, f32, f64
+
+    Supported Conversion Mode:
+      FP -> FP
+      INT -> FP
+      FP -> INT
+      INT -> INT
+  }];
+
+  let arguments = (ins
+    Tcp_Tensor:$in,
+    OptionalAttr<Tcp_SignednessAttr>:$in_int_signedness,
+    OptionalAttr<Tcp_SignednessAttr>:$out_int_signedness
```

Comment:
Have you considered an alternative design with multiple ops like in the Arith dialect? I.e. `sitofp`, `uitofp`, `fptosi`, `fptoui`, `extsi`, `extui`, `extf`, `trunci`, `truncf`?

Reply:
We debated about this. The summary was that we felt that is probably too low level of an abstraction for TCP. And it is better to represent "casting" as a single op so that transformations are simpler.

Let us know your thoughts on this. Are we missing anything obvious here?

Reply:
I can't say I have hard data on what approach should be preferred in this case. The optional signedness attributes felt a bit awkward, but having a bunch of ops instead of just one also has downsides. I think that the current design looks good for getting started and getting more experience with it.

Reply:
Sounds good. Lets keep this as one of the design points that needs to be revisited.